compiler/codegen/expr.cpp

```diff
@@ -1498,7 +1498,7 @@ GenRet codegenAdd(GenRet a, GenRet b)
       if(values.a->getType()->isFPOrFPVectorTy()) {
         ret.val = info->builder->CreateFAdd(values.a, values.b);
       } else {
-        ret.val = info->builder->CreateAdd(values.a, values.b, "", false, true);
+        ret.val = info->builder->CreateAdd(values.a, values.b, "", false, a_signed or b_signed);
```

Comment:
I thought it would be values.isSigned (which indicates whether whatever we promoted to is signed) and I don't think 'or' works in C++ (unless it is some new standard thing I've never seen?)